### PR TITLE
Fixes an invalid assert in No Payload Enum's collectArchetypeMetadata: the type itself might contain an Archetype. We need to check that it is not an Archetype itself

### PR DESCRIPTION
**Pull request by shajrawi**
radar rdar://problem/35831517

### CODE REVIEW

**Comment by shajrawi** [comment]
@swift-ci Please smoke test and merge

File: lib/IRGen/GenEnum.cpp [hunk 1]
```diff
@@ -945,7 +945,7 @@ namespace {
         llvm::MapVector<CanType, llvm::Value *> &typeToMetadataVec,
         SILType T) const override {
       auto canType = T.getSwiftRValueType();
-      assert(!canType->hasArchetype() &&
+      assert(!canType->getWithoutSpecifierType()->is<ArchetypeType>() &&
```
Comment (slavapestov) [inline lib/IRGen/GenEnum.cpp]: Do you have an example where getWithoutSpecifierType() is not a no-op?

**Comment by shajrawi** [comment]
@swift-ci Please smoke test and merge

**Comment by shajrawi** [comment]
@swift-ci Please smoke test and merge

**Pull request merged**